sycl/doc/OptionalDeviceFeatures.md

```diff
@@ -869,20 +869,123 @@ property set.
 
 #### Device configuration file
 
-The configuration file uses a simple YAML format where each top-level key is
-a name of a device architecture. There are sub-keys under each device for
-the supported aspects and sub-group sizes.  For example:
+The "device configuration file" is a [YAML](https://yaml.org) format file which
+stores information about all the AOT targets supported by the DPC++ compiler.
+The top-level keys (nodes) can be of 2 kinds: targets and property collections.
+- Targets are those strings which can appear in the `-fsycl-targets` list.
+- Property collections are basically named groups of properties to conveniently
+  refer to from other targets and property collections.
+
+> ##### Target name specification formats
+> Targets within can be specified in three forms: long, short and dynamic.
+> - Long form has the format `<vendor>`\_`<device type>`\_`<ID>`
+> - Short form has the format `<ID>`
+> - Dynamic form has the format `<vendor>`\_`<device type>`\_`*`. Dynamic
+>   targets are used for dynamically generated platforms, such as some Intel
+>   FPGA ones. More on this below.
+>
+> Some targets have only long form, some - both long and short. There is an aspect
+> with the same spelling for each long and short form target. There is no aspect
+> for dynamic targets.
```

Comment:
> Why are the lines above quoted (i.e. > prefix)?

do you mean `<` and `>` in `<vendor>` ? this is to illustrate that this is variable text rather than word "vendor". There should be no `<` and `>`  in actual vendor or device type or ID strings

> Later on in the spec we have the section "AOT target identification", which lists the actual target names for each device. Let's move any discussion about long-form vs. short-form to that section of the document.

ok, makes sense

> Therefore, I think we can assume that the DPC++ driver has read both the standard YAML file and the FPGA YAML file, so it will know the names of all devices.

I thought about this too. Maybe you are right. As long as clang++ ... -v shows all the device config files used, it is OK not to have everything listed in the main config.

Reply:
> do you mean < and > in <vendor> ?

No.  Each line from 879 to 889 starts with an initial ">" character.  This is markdown syntax for "quoted paragraph".  I don't see any reason why this paragraph should be a quote.

> I thought about this too. Maybe you are right. As long as clang++ ... -v shows all the device config files used, it is OK not to have everything listed in the main config.

I think this would be nicer because it simplifies the config file syntax.

Reply:
> No. Each line from 879 to 889 starts with an initial ">" character. This is markdown syntax for "quoted paragraph". I don't see any reason why this paragraph should be a quote.

I see. Yes, it was used to highlight as quoted, as this describes lower-level stuff (format). No problem to remove the quotes.

> I think this would be nicer because it simplifies the config file syntax.

ok